packages/flutter_test/lib/src/matchers.dart

```diff
@@ -1717,7 +1720,19 @@ class _MatchesGoldenFile extends AsyncMatcher {
 
   @override
   Description describe(Description description) =>
-      description.add('one widget whose rasterized image matches golden image "$key"');
+      description.add('one widget whose rasterized image matches golden image "${_getTestNameUri(key, version)}"');
+
+  Uri _getTestNameUri(Uri key, String version) {
+    return version == null ? key : Uri.parse(
+      key
+        .toString()
+        .splitMapJoin(
+        RegExp(r'.png'),
```

Comment:
Do we know that these will always be .png? What about using the path packages to just split off any extension?

Reply:
When we do the binding for a golden test, we have it write out in png format. See here: https://github.com/flutter/flutter/blob/dfecafa4ab30f9c509ce409241000cdd785ef23d/packages/flutter_test/lib/src/matchers.dart#L1702